src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs

```diff
@@ -198,4 +262,100 @@ private Task<ISyncedState<long>> CreateLastReadEntryState(Symbol chatId, Cancell
         }
         return listeningChatIds;
     }
+
+    private async Task<Symbol> GetListeningChatIdToEliminate(ImmutableList<Symbol> listeningChatIds, Symbol listenChatId, Symbol recordingChatId)
+    {
+        // When active chats panel limit exceeded, we look for a less important chat to eliminate it from the list.
+        var limit = ActiveChatsLimit;
+        if (!listenChatId.IsEmpty) {
+            if (!recordingChatId.IsEmpty && !listeningChatIds.Contains(recordingChatId) && listenChatId != recordingChatId)
+                limit--; // reserve one slot for recording chat
+        }
+
+        if (listeningChatIds.Count < limit)
+            return Symbol.Empty;
+
+        bool IsPinnedChat(Symbol c)
+            => PinnedChatIds.Value.Keys.Contains(c.Value, StringComparer.OrdinalIgnoreCase);
+
+        async Task<Symbol> GetChatToEliminate(ICollection<Symbol> chatIds)
```

Comment:
Let's put inner methods at the end of the parent method

Reply:
done